src/Services/MailFinderService.php

```diff
@@ -67,6 +82,25 @@ public function findTemplateByTechnicalName(
 
         $searchFolder = $this->searchPathProvider->buildPaths($businessEvent);
 
+        $criteria = (new Criteria)->addFilter(new EqualsFilter('salesChannels.id', $businessEvent->getSalesChannelId()));
+        $searchResult = $this->themeRepository->search($criteria, $businessEvent->getContext());
+
+        $criteria = (new Criteria)->addFilter(new EqualsFilter('name', $searchResult->first()->getTechnicalName()));
```

Comment:
Can you do this with plain sql?

Also an Theme can be an app

Reply:
@shyim we changed this to be plain sql and added support for the new feature from pull request #35 which we have not tested before.
Sadly it is not possible to have the correct path displayed in the Shopware Backend with our change, as this change makes the actual path related to the selected sales channel and theme (if the theme contains mail templates)